### DISCUSSION TITLE
baseURL and server/api

### DISCUSSION THREAD

**Discussion by kto32** [nuxt/nuxt#16707] [Questions]
- Operating System: `Linux`
- Node Version:     `v18.8.0`
- Nuxt Version:     `3.0.0-rc.11`
- Nitro Version:    `0.5.4`
- Package Manager:  `yarn@1.22.19`
- Builder:          `vite`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`

Reproduction: https://stackblitz.com/edit/github-dyhdgf-2ewcox?file=components/ForkTable.vue

Good afternoon, the project uses proxying via Nginx. Faced with such a problem that when setting the baseURL value to something other than the default value, the server/api stops working, is this a Nuxt3 error or my mistake? Help please.

If there are problems with the reproduction of the link, write to me to send the archive of the reproduction or update the link.

**Comment by kto32** [comment]
An interesting fact is if when requesting `$fetch('https://jsonplaceholder.typecode.com/todos/1')` pass the `baseURL` parameter equal to `"`, then everything works as it should, i.e. as a result it worked with `$fetch('https://jsonplaceholder.typecode.com/todos/1 ', { baseURL: " })`
I couldn't stand it in the reproduction.

**Comment by sevillaarvin** [comment]
Yeah the baseURL is set here: https://github.com/nuxt/framework/blob/f350a70775016a8bde92670b4856d5d1edb1e7e0/packages/nuxt/src/app/entry.ts#L18

So the workaround is as you mentioned.

Maybe create a feature request in [ohmyfetch](https://github.com/unjs/ohmyfetch) so that when fetch is given a url instead of a path, it will ignore the baseURL.

Though if I were you, for external apis I would use `$fetch("/todos/1", { baseURL: "https://jsonplaceholder.typecode.com" })`, or maybe create a new $fetch instance and then set the external baseURL there.

**Reply by kto32** [reply]
How much do you think the request in ohmyfetch is in demand?
At the moment, I did it on your recommendation, and yes, everything works fine.

**Reply by sevillaarvin** [reply]
> How much do you think the request in ohmyfetch is in demand?

I did a little digging and I think it's already fixed, but the new version is not yet released:

https://github.com/unjs/ufo/blob/501008eed395bc0376f5f47dc02876e25c34b057/src/utils.ts#L65
https://github.com/unjs/ufo/pull/68

**Reply by kto32** [reply]
Didn't the update come out? The merge seems to be completed (although marked as a bug), ufo v0.8.5 is contained in nuxt3, perhaps the dependency has not been updated in ohmyfetch?

**Reply by sevillaarvin** [reply]
Merged July 29 but 0.8.5 is released before that.
